src/hooks/useFormElementOptions.ts

```diff
@@ -90,11 +90,7 @@ export default function useFormElementOptions<T>({
       return
     }
 
-    if (
-      typeof value === 'string' &&
-      value &&
-      !withInjectedOptions.some((option) => value === option.value)
-    ) {
+    if (!withInjectedOptions.some((option) => value === option.value)) {
```

Comment:
I think we want to keep the `typeof value === 'string'` check here to avoid iterating over elements unnecessarily.